modules/hypelabBidAdapter.js

```diff
@@ -0,0 +1,159 @@
+  const media = data.creative_set.video || data.creative_set.image || {};
+  return { width: media.width, height: media.height };
+}
+
+function isBidRequestValid(bidderRequest) {
+  return (
+    !!bidderRequest.params?.property_slug &&
+    !!bidderRequest.params?.placement_slug
+  );
+}
+
+function buildRequests(validBidRequests, bidderRequest) {
+  const result = validBidRequests.map((request) => {
+    const uids = (request.userIdAsEids || []).reduce((a, c) => {
+      const ids = c.uids.map((uid) => uid.id);
+      return [...a, ...ids];
+    }, []);
+
+    const uuid = uids[0] ? uids[0] : generateTemporaryUUID();
+
+    const payload = {
+      property_slug: request.params.property_slug,
+      placement_slug: request.params.placement_slug,
+      provider_version: PROVIDER_VERSION,
+      provider_name: PROVIDER_NAME,
+      referrer:
+        bidderRequest.refererInfo?.page || typeof window != 'undefined'
+          ? window.location.href
+          : '',
+      sdk_version: PREBID_VERSION,
+      sizes: request.sizes,
+      wids: [],
+      uuid,
+      bidRequestsCount: request.bidRequestsCount,
+      bidderRequestsCount: request.bidderRequestsCount,
+      bidderWinsCount: request.bidderWinsCount,
+      wp: {
+        ada: typeof window != 'undefined' && !!window.cardano,
+        bnb: typeof window != 'undefined' && !!window.BinanceChain,
+        eth: typeof window != 'undefined' && !!window.ethereum,
+        sol: typeof window != 'undefined' && !!window.solana,
+        tron: typeof window != 'undefined' && !!window.tron,
+      },
+    };
+
+    return {
+      method: 'POST',
+      url: url(REQUEST_ROUTE),
+      options: { contentType: 'application/json', withCredentials: false },
+      data: payload,
+      bidId: request.bidId,
+    };
+  });
+
+  return result;
+}
+
+function generateTemporaryUUID() {
+  return 'tmp_' + generateUUID();
+}
+
+function interpretResponse(serverResponse, bidRequest) {
+  const { data } = serverResponse.body;
+
+  if (!data.cpm || !data.html) return [];
+
+  const size = mediaSize(data);
+
+  const result = {
+    requestId: bidRequest.bidId,
+    cpm: data.cpm,
+    width: size.width,
+    height: size.height,
+    creativeId: data.creative_set_slug,
+    currency: data.currency,
+    netRevenue: true,
+    referrer: bidRequest.data.referrer,
+    ttl: data.ttl,
+    ad: data.html,
+    mediaType: serverResponse.body.data.media_type,
+    meta: {
+      advertiserDomains: data.advertiserDomains || [],
+    },
+  };
+
+  return [result];
+}
+
+function getUserSyncs(syncOptions, serverResponses, gdprConsent, uspConsent) {
+  const syncs = [];
+  return syncs;
+}
+
+export function report(eventType, data, route = REPORTING_ROUTE) {
+  if (!route) return;
+
+  const options = {
+    method: 'POST',
+    contentType: 'application/json',
+    withCredentials: true,
+  };
+
+  const request = { type: eventType, data };
+  ajax(url(route), null, request, options);
```

Comment:
I'm not sure if a bid adapter is allowed to make external ajax calls. I'm checking with others on this but according to me, having an analytics adapter or a separate integration with an analytics provider is more appropriate.

Reply:
Got it. Let me know what you find. Here are a few other places where ajax POST requests are being used to record onBidWon and other callbacks:

https://github.com/prebid/Prebid.js/blob/222c702a469a861f03375be8800d3d85c996b36c/modules/adpartnerBidAdapter.js#L140

https://github.com/prebid/Prebid.js/blob/222c702a469a861f03375be8800d3d85c996b36c/modules/gnetBidAdapter.js#L107

https://github.com/prebid/Prebid.js/blob/222c702a469a861f03375be8800d3d85c996b36c/modules/datablocksBidAdapter.js#L190